packages/core/src/logger/logger.ts

```diff
@@ -1,144 +1,196 @@
     }
   }
 
-  fork<R>(callback: () => R): R;
-  fork<R>(force: true, callback: () => R): R;
-  fork<R>(callbackOrForce: boolean | (() => R), maybeCallback?: () => R): R {
-    const [callback, force] =
-      typeof callbackOrForce === 'boolean'
-        ? [maybeCallback!, callbackOrForce]
-        : [callbackOrForce, false];
+  public fork<R>(callback: () => R, force?: boolean): R {
+    const context = this.asyncStorage.getStore();
 
-    const tag = this.context.getStore();
-
-    if (tag && !force) {
+    if (context && !force) {
       throw new Error('Logger is already forked');
     }
 
-    return this.context.run(this.queueManager.createQueue(), callback);
-  }
-
-  debug(data: Record<string, any>): void;
-  debug(message: string, data?: Record<string, any>): void;
-  debug(message: string, params?: any[], data?: Record<string, any>): void;
-  debug(message: any, params?: any[], data?: Record<string, any>): void {
-    this.log(Logger.DEBUG, message, params, data);
-  }
-
-  info(data: Record<string, any>): void;
-  info(message: string, data?: Record<string, any>): void;
-  info(message: string, params?: any[], data?: Record<string, any>): void;
-  info(message: any, params?: any[], data?: Record<string, any>): void {
-    this.log(Logger.INFO, message, params, data);
-  }
-
-  warning(data: Record<string, any>): void;
-  warning(message: string, data?: Record<string, any>): void;
-  warning(message: string, params?: any[], data?: Record<string, any>): void;
-  warning(message: any, params?: any[], data?: Record<string, any>): void {
-    this.log(Logger.WARNING, message, params, data);
-  }
-
-  error(data: Record<string, any>): void;
-  error(message: string, data?: Record<string, any>): void;
-  error(message: string, params?: any[], data?: Record<string, any>): void;
-  error(message: any, params?: any[], data?: Record<string, any>): void {
-    this.log(Logger.ERROR, message, params, data);
-  }
-
-  log(level: number, data: Record<string, any>): void;
-  log(level: number, message: string, data?: Record<string, any>): void;
-  log(level: number, message: string, params?: any[], data?: Record<string, any>): void;
-  log(plugin: PluginId, level: number, data: Record<string, any>): void;
-  log(plugin: PluginId, level: number, message: string, data?: Record<string, any>): void;
-  log(
-    plugin: PluginId,
-    level: number,
-    message: string,
-    params?: any[],
-    data?: Record<string, any>,
-  ): void;
-  log(
-    pluginOrLevel: any,
-    levelOrMessage: any,
-    messageOrParams?: any,
-    paramsOrData?: any[],
-    maybeData?: Record<string, any>,
+    const envelopedCallback = () => {
+      const response = callback();
+      if (
+        typeof response === 'object' &&
+        typeof (response as unknown as Promise<any>).then === 'function'
+      ) {
+        (response as unknown as Promise<any>).then(() => {
+          this.flush();
+        });
+      } else {
+        this.flush();
```

Comment:
ha, už chápu, proč není potřeba explicitní `flush()`, nice! ale ještě by to chtělo ohandlit výjimky vyhozený callbackem, taky by měly zavolat `flush()` než výjimku vyhodí výš

Reply:
done